src/main/webapp/WEB-INF/classes/com/view/SignupServlet.java

```diff
@@ -35,18 +35,11 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
         // sendToProfileScreen and signupFailed
         this.request = request;
         this.response = response;
-
-        // get the username password and repeatedPassword from the input
-        // fields in the jsp file.
-        String username = request.getParameter("username");
-        String password = request.getParameter("password");
-        String repeatedPassword = request.getParameter("repeated password");
-
         // Initialize the loop for the use_case
         SignupController controller = SignupSetup.setup(this);
 
         // call the SignupController passing it the inputs
-        controller.execute(username, password, repeatedPassword);
+        controller.execute(request);
```

Comment:
Not sure if this is a good idea.

Reply:
Might be best to unpack it inside of the Servlet, I think we should keep request and response objects fixed internally.